### PR TITLE
add eq/ne extension for AnyRef|Null to Scala3RunTime

### PR DESCRIPTION
**Pull request by olhotak**
Addresses part 3 of #14622 .

### CODE REVIEW

File: library/src/scala/runtime/Scala3RunTime.scala [hunk 1]
```diff
@@ -19,4 +19,10 @@ object Scala3RunTime:
     if (isNull) throw new NullPointerException("tried to cast away nullability, but value is null")
     else x.asInstanceOf[x.type & T]
 
+  extension (inline x: AnyRef | Null)
```
Comment (smarter) [inline library/src/scala/runtime/Scala3RunTime.scala:22]: This needs to be in a minor release (e.g. 3.2) and then marked `@since("3.2")`

**Comment by olhotak** [comment]
This is failing CoverageTests:
```
Error:  Test dotty.tools.dotc.coverage.CoverageTests.checkCoverageStatements failed: java.lang.AssertionError: /tmp/coverage16107752629811232167/scoverage.coverage differs from expected /__w/dotty/dotty/tests/coverage/pos/Inlined.scoverage.check, took 1.075 sec
1039
```

@smarter what does this mean? How can I determine whether this is a bug or just that the `.check` file needs to be updated?

**Comment by smarter** [comment]
The test is in https://github.com/lampepfl/dotty/blob/main/tests/coverage/pos/Inlined.scala and the expected output in https://github.com/lampepfl/dotty/blob/main/tests/coverage/pos/Inlined.scoverage.check. I assume this is failing just because the line numbers of the assert macro changed, and testCompilation --update-checkfiles will fix it.

**Comment by olhotak** [comment]
It seems `testCompilation --update-checkfiles` doesn't run the coverage tests:
```
sbt:scala3> testCompilation --update-checkfiles
[info] Test run dotty.tools.dotc.CompilationTests started
[info] Test dotty.tools.dotc.CompilationTests.negAll started
[=======================================] completed (1443/1443, 0 failed, 78s)
[info] Test dotty.tools.dotc.CompilationTests.runAll started
[=======================================] completed (1470/1470, 0 failed, 110s)
[info] Test dotty.tools.dotc.CompilationTests.pickling started
[=======================================] completed (3760/3760, 0 failed, 92s)
[info] Test dotty.tools.dotc.CompilationTests.rewrites started
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (8/8, 0 failed, 0s)
[info] Test dotty.tools.dotc.CompilationTests.pos started
[=======================================] completed (2702/2702, 0 failed, 114s)
[info] Test dotty.tools.dotc.CompilationTests.checkInit started
[=======================================] completed (141/141, 0 failed, 1s)
[=======================================] completed (40/40, 0 failed, 0s)
[=======================================] completed (23/23, 0 failed, 1s)
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (1/1, 0 failed, 0s)
[=======================================] completed (1/1, 0 failed, 0s)
[info] Test dotty.tools.dotc.CompilationTests.posTwice started
[=======================================] completed (133/133, 0 failed, 20s)
[info] Test dotty.tools.dotc.CompilationTests.explicitNullsNeg started
[=======================================] completed (59/59, 0 failed, 4s)
[info] Test dotty.tools.dotc.CompilationTests.explicitNullsPos started
[=======================================] completed (74/74, 0 failed, 7s)
[info] Test dotty.tools.dotc.CompilationTests.explicitNullsRun started
[=======================================] completed (9/9, 0 failed, 0s)
[info] Test dotty.tools.dotc.CompilationTests.fuzzyAll started
[=======================================] completed (184/184, 0 failed, 3s)
[info] Test dotty.tools.dotc.CompilationTests.genericJavaSignatures started
[=======================================] completed (23/23, 0 failed, 0s)

================================================================================
Test Report
================================================================================

25 suites passed, 0 failed, 25 total
[info] Test run dotty.tools.dotc.CompilationTests finished: 0 failed, 0 ignored, 12 total, 441.386s
[info] Test run dotty.tools.dotc.BootstrappedOnlyCompilationTests started
[info] Test run dotty.tools.dotc.BootstrappedOnlyCompilationTests finished: 0 failed, 0 ignored, 0 total, 0.0s
[info] Test run dotty.tools.dotc.coverage.CoverageTests started
[info] Test run dotty.tools.dotc.coverage.CoverageTests finished: 0 failed, 0 ignored, 0 total, 0.0s
[info] Passed: Total 12, Failed 0, Errors 0, Passed 12
[success] Total time: 443 s (07:23), completed May 9, 2022, 1:29:19 p.m.
sbt:scala3> 
```

**Comment by smarter** [comment]
Sorry, the tests are only run with the bootstrapped compiler so you need `scala3-compiler-bootstrapped/testCompilation --update-checkfiles`

**Review by smarter — APPROVED** [review]
Could you add some documentation comments for these methods? Otherwise LGTM.

File: library/src/scala/runtime/stdLibPatches/Predef.scala [hunk 2]
```diff
@@ -47,4 +49,16 @@ object Predef:
    */
   extension [T](x: T | Null) inline def nn: x.type & T =
     scala.runtime.Scala3RunTime.nn(x)
+
+  /** Enables an expression of type `T|Null`, where `T` is a subtype of `AnyRef`, to be checked for `null`
+   *  using `eq` and `ne`, rather than only `==` and `!=`. This is needed because `Null` no longer has
+   *  `eq` or `ne` methods, only `==` and `!=` inherited from `Any`. */
```
Comment (smarter) [inline library/src/scala/runtime/stdLibPatches/Predef.scala:55]: I think comments need to be on individual methods, not sure if scaladoc/IDEs pick up anything from the extension block.

**Pull request merged**